### PR TITLE
Remove inforgraphic in favor of in-app help

### PR DESCRIPTION
**Pull request by mmcfarland**
#### Overview
Plugins can register with the framework to acknowledge that there is
in-app help functionality available.  The framework handles the UI of
showing the help icon in the header and for managing the state of "show on
start" via local storage.  Its intended purpose is to allow plugins to
manage their own help functionality and move away from the exising
infographic approach.

The details of the approach should match the associated issue, which is descriptive

Connects #782 

#### Testing
* Load the Itendify Point plugin, verify that help is shown.
* Dismiss the in-app help, and click the `i` info button in the header, verify help is shown again (dismiss)
* Close plugin and re-open, verify that the help is not shown
* Refresh the page and do the same.

The goal is to allow the plugin to handle some in app instruction, with a light UI mechanism from the framework and the ability to manage whether the app shows help upon activation.  It's meant to be flexible, and the above is just one workflow that could accommodate it.

### CODE REVIEW

**Comment by jtarricone** [comment]
I noticed the help message might display in an unintended case. After switching to the branch, the app behaved as expected in the testing instructions. Upon page refresh and hard refresh, the help text was not displayed. Switching to another browser and loading it for the first time (in that browser), the help text was not displayed. I think this is the desired behavior.

However, selecting the "Identify" plugin (w/o help message) and then activating another plugin, then switching back brought the message back, but it didn't seem to be persisted in the sense that another browser didn't show the message when the plugin was activated.

**Comment by jtarricone** [comment]
In IE11, clicking "Got It" to get rid of the help message resulted in a scrollbar showing on the right of the sidebar.

File: src/GeositeFramework/sample_plugins/identify_point/main.js [hunk 1]
```diff
@@ -43,6 +53,28 @@ define(["dojo/_base/declare", "framework/PluginBase"],
...
+
+                self.$helpMsg = $("<div>Help: Click the map to see the point </div>");
+
+                $("<a>", {
+                    className: "button",
+                    href: '#',
+                    text: "Got it",
+                    click: function() {
+                        self.$helpMsg.remove();
+                        this.remove();
```
Comment (jtarricone) [inline src/GeositeFramework/sample_plugins/identify_point/main.js]: This throws when evaluated in IE11.

File: src/GeositeFramework/js/Plugin.js [hunk 2]
```diff
@@ -46,7 +46,8 @@ require(['use!Geosite',
                         _unsafeMap: esriMap,
                         downloadAsCsv: requestCsvDownload,
                         downloadAsPlainText: requestTextDownload,
-                        dispatcher: N.app.dispatcher
+                        dispatcher: N.app.dispatcher,
+                        supressHelpOnStartup: _.partial(supressHelpOnStartup, model)
```
Comment (jtarricone) [inline src/GeositeFramework/js/Plugin.js]: Not consequential, but "suppress" is misspelled.

**Comment by jtarricone** [comment]
It looks like any toggling of the plugin is enough to reactivate the help message, whether it's just (de)-activating the same one or switching to another one and then back. Is the Identify plugin supposed to have the behavior described in #782 or is that one not intended to persist the setting?

**Comment by mmcfarland** [comment]
Updated with fix for toggle bug and spelling errors.  The error in IE was vestigial code, so I removed it and I'm not concerned about the sidebar scroll since this is just a proof of concept example and probably will never be run in IE.

If you want to "unset" the suppress flag, you can do so in the devtools Application tab:
screenshot from 2016-12-20 14 17 41

**Pull request merged**